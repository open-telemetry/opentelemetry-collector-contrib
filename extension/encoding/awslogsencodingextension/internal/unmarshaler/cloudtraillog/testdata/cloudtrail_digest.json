{
  "awsAccountId": "111122223333",
  "digestStartTime": "2025-10-21T00:00:00Z",
  "digestEndTime": "2025-10-21T01:00:00Z",
  "digestS3Bucket": "amzn-s3-demo-bucket",
  "digestS3Object": "AWSLogs/111122223333/CloudTrail-Digest/us-east-1/2025/10/21/digest.json.gz",
  "digestPublicKeyFingerprint": "31e8b5433410dfb61a9dc45cc65b22ff",
  "digestSignatureAlgorithm": "SHA256withRSA",
  "newestEventTime": "2025-10-21T00:59:58Z",
  "oldestEventTime": "2025-10-21T00:00:01Z",
  "previousDigestS3Bucket": "amzn-s3-demo-bucket",
  "previousDigestS3Object": "AWSLogs/123456789012/CloudTrail-Digest/us-east-1/2025/10/21/digest-previous.json.gz",
  "previousDigestHashValue": "97fb791cf91ffc440d274f8190dbdd9aa09c34432aba82739df18b6d3c13df2d",
  "previousDigestHashAlgorithm": "SHA-256",
  "previousDigestSignature": "hash-of-previous-digest",
  "logFiles": [
    {
      "s3Bucket": "amzn-s3-demo-bucket",
      "s3Object": "AWSLogs/123456789012/CloudTrail/us-east-1/2025/10/21/obj1.json.gz",
      "hashValue": "7aab8a1f04f32adf1113c9acdd894f10a2b5fa4039fcb4dd1cbd40b13e482b8a",
      "hashAlgorithm": "SHA-256",
      "newestEventTime": "2025-10-21TT00:52:27Z",
      "oldestEventTime": "2025-10-21TT00:42:27Z"
    }
  ]
}