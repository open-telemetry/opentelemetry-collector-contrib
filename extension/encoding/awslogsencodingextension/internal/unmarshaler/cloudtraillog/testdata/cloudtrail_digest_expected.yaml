resourceLogs:
  - resource: {}
    scopeLogs:
      - logRecords:
          - attributes:
              - key: cloud.account.id
                value:
                  stringValue: "111122223333"
              - key: aws.cloudtrail.digest.end_time
                value:
                  stringValue: "2025-10-21T01:00:00Z"
              - key: aws.cloudtrail.digest.s3_bucket
                value:
                  stringValue: amzn-s3-demo-bucket
              - key: aws.cloudtrail.digest.s3_object
                value:
                  stringValue: AWSLogs/111122223333/CloudTrail-Digest/us-east-1/2025/10/21/digest.json.gz
              - key: aws.cloudtrail.digest.newest_event
                value:
                  stringValue: "2025-10-21T00:59:58Z"
              - key: aws.cloudtrail.digest.oldest_event
                value:
                  stringValue: "2025-10-21T00:00:01Z"
              - key: aws.cloudtrail.digest.previous_s3_bucket
                value:
                  stringValue: amzn-s3-demo-bucket
              - key: aws.cloudtrail.digest.previous_s3_object
                value:
                  stringValue: AWSLogs/123456789012/CloudTrail-Digest/us-east-1/2025/10/21/digest-previous.json.gz
              - key: aws.cloudtrail.digest.log_files
                value:
                  arrayValue:
                    values:
                      - kvlistValue:
                          values:
                            - key: s3_bucket
                              value:
                                stringValue: amzn-s3-demo-bucket
                            - key: s3_object
                              value:
                                stringValue: AWSLogs/123456789012/CloudTrail/us-east-1/2025/10/21/obj1.json.gz
                            - key: newest_event_time
                              value:
                                stringValue: 2025-10-21TT00:52:27Z
                            - key: oldest_event_time
                              value:
                                stringValue: 2025-10-21TT00:42:27Z
            body: {}
            timeUnixNano: "1761004800000000000"
        scope:
          attributes:
            - key: encoding.format
              value:
                stringValue: aws.cloudtrail
          name: github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension
          version: test-version
